wizard/WizardCreateWalletFromDevice.qml

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2014-2018, The Monero Project
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import QtQuick 2.2
+import moneroComponents.WalletManager 1.0
+import moneroComponents.Wallet 1.0
+import QtQuick.Layouts 1.1
+import QtQuick.Dialogs 1.2
+import 'utils.js' as Utils
+
+ColumnLayout {
+    opacity: 0
+    visible: false
+
+    Behavior on opacity {
+        NumberAnimation { duration: 100; easing.type: Easing.InQuad }
+    }
+
+
+    onOpacityChanged: visible = opacity !== 0
+
+    function onWizardRestarted() {
+        // reset account name field
+        uiItem.accountNameText = defaultAccountName
+    }
+
+    //! function called each time we display this page
+
+    function onPageOpened(settingsOblect) {
+        checkNextButton()
+    }
+
+    function onPageClosed(settingsObject) {
+        settingsObject['account_name'] = uiItem.accountNameText
+        settingsObject['wallet_path'] = uiItem.walletPath
+        var restoreHeight = parseInt(uiItem.restoreHeight);
+        settingsObject['restore_height'] = isNaN(restoreHeight)? 0 : restoreHeight;
```

Comment:
That might need double checking, it's a bit wtf

Reply:
It does look a bit wtf, but generally I have very little idea about how things should be done QML, and this piece of code was mostly copied from WizardRecoveryWallet.qml:

https://github.com/monero-project/monero-gui/blob/cd46edb23f0764b7bb7ee1171224881554be5d2f/wizard/WizardRecoveryWallet.qml#L70

I think someone who knows QML better should handle these issues in the future refactoring.